File: app/src/main/java/com/szollosi/sensormusicplayer/fragments/FragmentXYZ.java
package com.szollosi.sensormusicplayer.fragments;

import android.os.Bundle;
import android.os.Handler;
import android.os.Looper;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.github.mikephil.charting.charts.LineChart;
import com.github.mikephil.charting.data.LineData;
import com.szollosi.sensormusicplayer.Constants;
import com.szollosi.sensormusicplayer.R;

public class FragmentXYZ extends BaseFragment {

  private LineChart chart;

  private Handler mHandler;

  private Runnable refreshGraph = new Runnable() {

    @Override
    public void run() {
      synchronized (dataPointsX) {
        seriesX.setValues(dataPointsX);
        seriesY.setValues(dataPointsY);
        seriesZ.setValues(dataPointsZ);

        chart.setData(new LineData(seriesX, seriesY, seriesZ));
        chart.invalidate();
        mHandler.postDelayed(this, Constants.DELAY_REFRESH);
      }
    }
  };

  private int page;

  private String title;


  @Nullable
  @Override
  public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
      @Nullable Bundle savedInstanceState) {
    View rootView = inflater.inflate(R.layout.fragment_section_xyz, container, false);
    chart = rootView.findViewById(R.id.graph_xyz);
    return rootView;
  }

  public static FragmentXYZ newInstance(int page, String title) {
    FragmentXYZ fragmentFirst = new FragmentXYZ();
    Bundle args = new Bundle();
    args.putInt("someInt", page);
    args.putString("someTitle", title);
    fragmentFirst.setArguments(args);
    return fragmentFirst;
  }

  @Override
  public void onResume() {
    super.onResume();
    mHandler = new Handler(Looper.myLooper());
    mHandler.postDelayed(refreshGraph, Constants.DELAY_REFRESH);
  }

  @Override
  public void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    page = getArguments().getInt("someInt", 0);
    title = getArguments().getString("someTitle");
  }

  @Override
  public void onPause() {
    super.onPause();
    mHandler.removeCallbacks(refreshGraph);
  }
}

File: app/src/main/java/com/szollosi/sensormusicplayer/fragments/FragmentXYZ.kt
package com.szollosi.sensormusicplayer.fragments

import android.os.Bundle
import android.os.Handler
import android.os.Looper
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

import com.github.mikephil.charting.charts.LineChart
import com.github.mikephil.charting.data.LineData
import com.szollosi.sensormusicplayer.MyConstants
import com.szollosi.sensormusicplayer.R

class FragmentXYZ : BaseFragment() {

    private var chart: LineChart? = null

    private var mHandler: Handler? = null

    private val refreshGraph = object : Runnable {

        override fun run() {
            synchronized(dataPointsX) {
                seriesX.values = dataPointsX.toArrayList()
                seriesY.values = dataPointsY.toArrayList()
                seriesZ.values = dataPointsZ.toArrayList()

                chart!!.data = LineData(seriesX, seriesY, seriesZ)
                chart!!.invalidate()
                mHandler!!.postDelayed(this, MyConstants.DELAY_REFRESH.toLong())
            }
        }
    }

    private var page: Int = 0

    private var title: String? = null


    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        val rootView = inflater.inflate(R.layout.fragment_section_xyz, container, false)
        chart = rootView.findViewById(R.id.graph_xyz)
        return rootView
    }

    override fun onResume() {
        super.onResume()
        Looper.myLooper()?.let {
            mHandler = Handler(it).apply {
                postDelayed(refreshGraph, MyConstants.DELAY_REFRESH.toLong())
            }
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        arguments?.let {
            page = it.getInt("someInt", 0)
            title = it.getString("someTitle")
        }
    }

    override fun onPause() {
        super.onPause()
        mHandler?.removeCallbacks(refreshGraph)
    }

    companion object {

        fun newInstance(page: Int, title: String): FragmentXYZ {
            val fragmentFirst = FragmentXYZ()
            val args = Bundle()
            args.putInt("someInt", page)
            args.putString("someTitle", title)
            fragmentFirst.arguments = args
            return fragmentFirst
        }
    }
}
